azkaban-exec-server/src/main/java/azkaban/execapp/ProjectVersion.java

```diff
@@ -16,34 +16,29 @@
 
 package azkaban.execapp;
 
+import com.google.common.base.Preconditions;
 import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.StandardCopyOption;
-import java.util.zip.ZipFile;
-
-import org.apache.commons.io.FileUtils;
 import org.apache.log4j.Logger;
 
-import azkaban.project.ProjectFileHandler;
-import azkaban.project.ProjectLoader;
-import azkaban.project.ProjectManagerException;
-import azkaban.utils.FileIOUtils;
-import azkaban.utils.Utils;
+import static com.google.common.base.Preconditions.*;
+
 
 public class ProjectVersion implements Comparable<ProjectVersion> {
   private final int projectId;
   private final int version;
+
   private File installedDir;
 
   public ProjectVersion(int projectId, int version) {
+    checkArgument(projectId > 0);
+    checkArgument(version > 0);
+
     this.projectId = projectId;
     this.version = version;
   }
```

Comment:
I have thought about this. But I want to push this for later. This class is heavily used and I didn't want to alter the flow in the first change.